<launch>

    <arg name="target_pose_x" default="1" />
    <arg name="target_pose_y" default="1" />
    <arg name="target_pose_z" default="1" />

    <!-- By default, we do not start a database (it can be large) -->
    <arg name="db" default="false" />
    <!-- Allow user to specify database location -->
    <arg name="db_path" default="$(find stretch_moveit_config)/default_warehouse_mongo_db" />

    <!-- By default, we are not in debug mode -->
    <arg name="debug" default="false" />

    <!--
  By default, hide joint_state_publisher's GUI

  MoveIt's "demo" mode replaces the real robot driver with the joint_state_publisher.
  The latter one maintains and publishes the current joint configuration of the simulated robot.
  It also provides a GUI to move the simulated robot around "manually".
  This corresponds to moving around the real robot without the use of MoveIt.
  -->
    <arg name="use_gui" default="false" />

    <!-- Gazebo specific options -->
    <arg name="gazebo_gui" default="true" />
    <arg name="paused" default="false" />
    <!-- By default, use the urdf location provided from the package -->
    <arg name="urdf_path" default="$(find mobiman)/urdf/stretch_gazebo.urdf.xacro" />

    <include file="$(find stretch_gazebo)/launch/gazebo.launch">
        <arg name="rviz" value="false" />
        <arg name="model" value="$(find mobiman)/urdf/stretch_gazebo.urdf.xacro"/>\
        <arg name="world" value="$(find stretch_gazebo)/worlds/random_objects.world" />
    </include>

    <!-- Load the URDF, SRDF and other .yaml configuration files on the param server -->
    <include file="$(find mobiman)/launch//moveit_config/planning_context.launch">
        <arg name="load_robot_description" value="false" />
    </include>

    <!-- If needed, broadcast static tf for robot root -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="virtual_joint_broadcaster_0" args="0 0 -0.01 0 0 0 base_link virtual_world_link" />

    <include file="$(find mobiman)/launch/move_group.launch">
        <arg name="allow_trajectory_execution" value="true" />
        <arg name="info" value="true" />
        <arg name="debug" value="$(arg debug)" />
    </include>


    <!-- Run Rviz and load the default config to see the state of the move_group node -->
    <include file="$(find mobiman)/launch/moveit_config/moveit_rviz.launch">
        <arg name="rviz_config" value="$(find mobiman)/launch/moveit_config/moveit.rviz" />
        <arg name="debug" value="$(arg debug)" />
    </include>
    
    <include file="$(find mobiman)/launch/move_base.launch">
    </include>


    <!-- include c++ script to do motion planning-->
    <node pkg="mobiman" name="moveit_demo" type="moveit_demo" output="screen">
        <param name="target_pose_x" value="$(arg target_pose_x)" />
        <param name="target_pose_y" value="$(arg target_pose_y)" />
        <param name="target_pose_z" value="$(arg target_pose_z)" />
    </node>

</launch>